azure-application-insights-spring-boot-starter/src/main/java/com/microsoft/applicationinsights/boot/ApplicationInsightsProperties.java

```diff
@@ -103,6 +128,114 @@ public void setLogger(Logger logger) {
         this.logger = logger;
     }
 
+    public static class Channel {
```

Comment:
@gavlyukovskiy I think we need default values here. If the user does not specify this property the default InProcessTelemetryChannel should be created. Does `@ConditionalOnMissingBean` annotation you added address this, in case user doesn't specify have this property in his properties file will the default channel get created (I hope so it does based on Line 85 in `TelemetryConfigurationFactory` Enum). However if the user does specify this property then you are using a parametrized constructor which implies that user has to now provide all the values and has lost the ability to tweak single property of channel say turn on developer mode or change end point address. Please correct me if I am wrong here, but if that is indeed the case we need to make sure there are default values for this and they should match default values of `InProcessTelemetryChannel` class.

Further we also need to set conditionals on the value the user specifies. I know that limit enforcer does this but setting conditions here would also be very useful. What do you think ?

Reply:
I added default values for every property in latest commit. Yes, I added `@ConditionalOnMissingBean` for user to be able to override this bean if needed. Even now with all properties configurable there are could be use-cases for that, for example using custom channel that will log every telemetry into the log or something like this.

Reply:
What are the conditions you're asking? Something like that the maximum buffer size should be between 1 and 1000?

Reply:
@gavlyukovskiy Yes exactly this is what I meant. If user uses inprocess telemetry channel and specifies different values then what channel expects then we should take the default value. As I mentioned that there is a logic inside the channel using `LimitsEnforcer` class however in order to have our starter to be very self sufficient and robust I think we should have this checks here. What do you think?